### PR TITLE
[SPARK-46560][PYTHON][DOCS] Refine docstring `reverse/map_contains_key`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to refine docstring of `reverse/map_contains_key`.

### Why are the changes needed?
To improve PySpark documentation.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manually test.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: python/pyspark/sql/functions/builtin.py [hunk 1]
```diff
@@ -15313,18 +15313,23 @@ def reverse(col: "ColumnOrName") -> Column:
...
     Returns
     -------
     :class:`~pyspark.sql.Column`
-        array of elements in reverse order.
+        A new column that contains a reversed string or an array with elements in reverse order.
 
     Examples
     --------
+    Example 1: Reverse a string
+
```
Comment (LuciferYang) [inline python/pyspark/sql/functions/builtin.py:15326]: Let's standardize the format:
1. import functions as sf and import other necessaries
2. try not to use `.alias`
3. perhaps `show` is clearer than `collect` when it comes to displaying results.

Comment (panbingkun) [reply python/pyspark/sql/functions/builtin.py:15326]: Done

File: python/pyspark/sql/functions/builtin.py [hunk 2]
```diff
@@ -15427,15 +15443,20 @@ def map_contains_key(col: "ColumnOrName", value: Any) -> Column:
...
     +-------------------------+
     |map_contains_key(data, 1)|
     +-------------------------+
     |                     true|
     +-------------------------+
-    >>> df.select(map_contains_key("data", -1)).show()
+
+    Example 2: The key is not in the map
+
+    >>> df.select(sf.map_contains_key("data", -1)).show()
```
Comment (LuciferYang) [inline python/pyspark/sql/functions/builtin.py:15459]:
```suggestion
    >>> from pyspark.sql import functions as sf
    >>> df = spark.sql("SELECT map(1, 'a', 2, 'b') as data")
    >>> df.select(sf.map_contains_key("data", -1)).show()
```

File: python/pyspark/sql/functions/builtin.py [hunk 3]
```diff
@@ -15313,21 +15313,37 @@ def reverse(col: "ColumnOrName") -> Column:
...
+    +-------------+
+    |    LQS krapS|
+    +-------------+
+
+    Example 2: Reverse an array
+
     >>> df = spark.createDataFrame([([2, 1, 3],) ,([1],) ,([],)], ['data'])
-    >>> df.select(reverse(df.data).alias('r')).collect()
-    [Row(r=[3, 1, 2]), Row(r=[1]), Row(r=[])]
+    >>> df.select(sf.reverse(df.data)).show()
```
Comment (LuciferYang) [inline python/pyspark/sql/functions/builtin.py:15339]:
```suggestion
    >>> from pyspark.sql import functions as sf
    >>> df = spark.createDataFrame([([2, 1, 3],) ,([1],) ,([],)], ['data'])
    >>> df.select(sf.reverse(df.data)).show()
```

**Review by itholic — APPROVED** [review]

File: python/pyspark/sql/functions/builtin.py (hunk 2 above)
Comment (panbingkun) [reply python/pyspark/sql/functions/builtin.py:15459]: Do we need to make each example self-contained now?
I remember writing this way before, it was suggested not to write duplicate but reusable code in every example.

Comment (LuciferYang) [reply python/pyspark/sql/functions/builtin.py:15459]: I prefer to `self-contained`, cc @zhengruifeng

Comment (panbingkun) [reply python/pyspark/sql/functions/builtin.py:15459]: Actually, I can do it all, but I think it's best for us to have `a stable rule`.

Comment (dongjoon-hyun) [reply python/pyspark/sql/functions/builtin.py:15459]: cc @HyukjinKwon for the above question about `from pyspark.sql import functions as sf`
> Do we need to make each example self-contained now?

Comment (HyukjinKwon) [reply python/pyspark/sql/functions/builtin.py:15459]: yeah. we have discussed in some other PRs. Should make it self-contained with `from pyspark.sql import functions as sf`.

Comment (dongjoon-hyun) [reply python/pyspark/sql/functions/builtin.py:15459]: Thank you for informing us the guidance, @HyukjinKwon . cc @panbingkun

**Review by dongjoon-hyun — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

File: python/pyspark/sql/functions/builtin.py [hunk 4]
```diff
@@ -15313,21 +15313,39 @@ def reverse(col: "ColumnOrName") -> Column:
...
+    +-------------+
+    |reverse(data)|
+    +-------------+
+    |    LQS krapS|
+    +-------------+
+
+    Example 2: Reverse an array
+
+    >>> df = spark.createDataFrame([([2, 1, 3],) ,([1],) ,([],)], ['data'])
```
Comment (zhengruifeng) [inline python/pyspark/sql/functions/builtin.py:15338]: nit, is this line duplicated?

Comment (panbingkun) [reply python/pyspark/sql/functions/builtin.py:15338]: For `self-contained`, the data is also repeated once in each example.

Comment (zhengruifeng) [reply python/pyspark/sql/functions/builtin.py:15338]: I mean here

https://github.com/apache/spark/pull/44561/files?diff=split&w=0#diff-52f85aab7bad664787b9ac7b86cdf1871e2a51bd05059841fcab67b662449095R15339-R15341

Comment (panbingkun) [reply python/pyspark/sql/functions/builtin.py]: Oh, I made a mistake. It was defined twice in this example and let me to remove it.
Done.

**Pull request closed without merging**